danceoval/search-and-destroy
# Dependency and Formatting Updates

## Need
- Remove dependency vulnerability
- Inconsistent formatting throughout (tabs/spaces, missing spacing in arrow functions, semicolons)

## Implementation
- [x] Updates version of `mocha` to remove vulnerability
- [x] Makes formatting consistent throughout
  - semicolons
  - spacing (2)
- [x] `findLoop` extra credit function
  - scaffolds function
  - adds it as an export
  - requires it in in the associated test file
- [x] Emphasizes how to run tests in the `README.md` by adding an "Important" section title
- [x] Updates tests to use functional expressions instead of arrow functions
  - Reason: Arrow functions _cannot_ be used when the `this` context matters. **Although**, `this` does ***not*** matter in this case, I believe it is best to keep all tests consistent throughout the curriculum to avoid confusion. (Reference: [Using ES6 arrow functions in Mocha](https://remarkablemark.org/blog/2016/08/20/arrow-functions-in-mocha/))

## Steps to test
1. `cd` into each sub-directory (`binary-search`, `looper`, and `min-jumps`)
2. Run `npm install`
3. Run `npm test` to make sure the tests run smoothly (they should fail)

## Related PRs
- TODO (solution branch)
